client/client.go

```diff
@@ -281,21 +280,131 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
-func dialOpenSSL(conf Config, network string, addr string) (net.Conn, error) {
-	contextSSL, err := openssl.NewCtx()
-	// probably should consider reusing the context, but then we
-	// have to propagate it with every request.
-	err = contextSSL.LoadVerifyLocations(conf.ServerCert, "")
+func listSystemCertsFound(certDir string) (int, error) {
+	sysCertsFound := 0
+	files, err := ioutil.ReadDir(certDir)
 	if err != nil {
-		return nil, err
+		return 0, fmt.Errorf("Failed to read the OpenSSL default directory (%s). Err %v", certDir, err.Error())
 	}
+	for _, certFile := range files {
+		certBytes, err := ioutil.ReadFile(path.Join(certDir, certFile.Name()))
+		if err != nil {
+			log.Debugf("Failed to read the certificate file for the HttpsClient. Err %v", err.Error())
+			continue
+		}
+
+		certs := openssl.SplitPEM(certBytes)
+		if len(certs) == 0 {
+			log.Debugf("No PEM certificate found in '%s'", certFile)
+			continue
+		}
+		first, certs := certs[0], certs[1:]
+		_, err = openssl.LoadCertificateFromPEM(first)
```

Comment:
Should it not load them all? I'm no expert on how System Certs are stored, but I can imagine they could contain concatenated certificates.

Reply:
If I've understood this correctly, this is done automatically from the `https://www.openssl.org/docs/man1.1.0/man3/SSL_CTX_set_default_verify_paths.html`, and `http://www.openssl.org/docs/ssl/SSL_CTX_load_verify_locations.html` functions.